### ISSUE TITLE
cibersortx issues

### ISSUE THREAD

**Issue by grst** [omnideconv/omnideconv#103]
I somehow can't get to work cibersortx locally when using one of the following options

 * `--absolute TRUE`
 * `--rmbatchSmode TRUE`. 

In both cases, the resulting cell-type matrix contains only zeros (or values very, very close to zero, like 1e-12). 

Using `--abs_method "no.sumto1"` solves the problem, but only if no batch correction is enabled. Using the same options in the web portal works without issues. It seems like the docker container does not match the version in the web portal. Last update of the container was 3 years ago. 

Can anyone confirm this?

**Comment by alex-d13** [comment]
So I was spending some time on this today.

I cannot really replicate your issue with the 0s in the cell-type matrix; do you remember what dataset you were using there?

But I can confirm, that the results of cibersortx web and docker are different, even for our small omnideconv example datasets:
![image](https://user-images.githubusercontent.com/40594032/190230857-1136221c-a894-4df2-85b4-70553d3b6543.png)


Of course, also the signatures look different, but i did not come up with a nice visualization of the fly. Even the number of genes in the signatures differ (1838 in Docker, 1888 in Web).

These are not really great news, because as Gregor mentioned, the Docker image is now about 2 years old, so we will most likely benchmark an old version of cibersortx.
Should we write the authors and attach this small example to ask them to update their Docker image? 

Just tagging the team here for input @omnideconv/core-team

**Comment by FFinotello** [comment]
Hey @alex-d13 thank you so much for looking into this. Could it be that the default parameters settings are different in these two implementations? Do we have a way to check this?

**Comment by FFinotello** [comment]
See also this discussion: https://www.biostars.org/p/428905/

**Comment by alex-d13** [comment]
> Could it be that the default parameters settings are different in these two implementations?

As far as the documentation of the docker image and the web verison goes, they use the same default parameters for signature building and deconvolution.

**Comment by alex-d13** [comment]
I also now compared the docker and web version to the true reference cell type fractions:
![image](https://user-images.githubusercontent.com/40594032/190479206-32f88f8c-bb7f-4f04-8f37-6cae5c71b176.png)
![image](https://user-images.githubusercontent.com/40594032/190479337-af6a7226-19e0-4d7b-8399-c727e26628e3.png)

**Comment by grst** [comment]
> I cannot really replicate your issue with the 0s in the cell-type matrix; do you remember what dataset you were using there?

It was related to the lung cancer atlas, maybe I can dig it out end of next week when I'm back from vacation. 

IIRC it occurred only with the s-mode batch correction.

**Comment by grisaitis** [comment]
hi all! i just learned about this repo and saw this issue. i have used cibersortx quite a bit so i think i might be able to help.

@alex-d13, since you are computing a signature matrix each time, i bet these differences are due to random number generation.

the paper says this about making a signature matrix from scRNA-seq (emphasis added):
> For each cell type represented by at least three single cells, *we selected 50% of all available single-cell GEPs using random sampling without replacement* (fractional sample sizes were rounded up such that two cells were sampled if only three were available). We then aggregated the profiles by summation in non-log linear space and normalized each population-level GEP into TPM. This process was repeated in order to generate five aggregated transcriptome replicates per cell type.

if that hypothesis is correct, you should see identical results if you re-run the job using the same signature matrix on each platform.

i think the second most likely possibility is the parameters. CIBERSORTx prints out all params its given in the first few lines of log output. this is visible for every job on the web app, btw. 

for example:
```
>Running CIBERSORTx GEP imputation (representative profiles only)...
>[Options] mixture: files/<your registration email>/filename
>[Options] sigmatrix: files/<your registration email>/filename
>[Options] rmbatchBmode: TRUE
>[Options] sourceGEPs: files/<your registration email>/filename
>[Options] threads: 7
>[Options] nsampling: 30
>[Options] redo: TRUE [to rerun CIBERSORT, set 'redocibersort' to TRUE]
>[Options] QN: FALSE
>[Options] outdir: files/william_grisaitis@dfci.harvard.edu/results/
>[Options] label: Job10
>Estimating cell proportions...
```

so in short, what i think i would do is:
1. run things on the web app:
   - run "create signature matrix", download the signature matrix file
   - run "impute cell fractions", download the log file and the results
2. locally,
   - from the log file, parse out what the exact parameters were on the web app
   - run the docker fractions image with those params (and the downloaded signature matrix file)
   - (maybe also run without specifying params, and check if the param values listed in the local log output are identical to the logs from the web app)
3. compare results
 
if anyone else knows of other differences, i would love to know!
